### ISSUE TITLE
resolve should take a Path instead of a Request (IMO)

### ISSUE THREAD

**Issue by rrichardson** [stephank/hyper-staticfile#18]
First off, thanks for making the library!  I was able to integrate this into a proxy that we deploy to customer sites and get huge speed improvements by pre-caching files then serving them with staticfile!  So.. great work! 

One minor nit that definitely tripped me up.  It is documented, but I missed it,  IMO we should use types to enforce this.  

Presently,  `resolve` takes a Request, but if there is anything in the Request except a path Part, it fails at runtime, and in a sort of a confusing way. 

In addition, constructing an http::Request that only contains a path is actually a bit of a PITA.  I haven't found an ergonomic way to create them.  

My proposal is to make `resolve` take 2 path`Path` or a `AsRef<Path>` parameter,  then there can be no ambiguity about what kind of data `resolve` expects. 

This would also streamline the resolve function as well, since it wouldn't need to perform the checks on the 'request` and unpack the uri().path() into a Path, it would already be done. 

The ergonomics actually get better for the user.  Instead of having to do this  (Note that I was not able to make `Request::get('/path/to/file')` work.. it failed as an invalid URI) 

```
let hpath: http::uri::PathAndQuery = filename.parse().unwrap();
let mut parts = http::uri::Parts::default();
parts.path_and_query = Some(hpath);
let uri = http::uri::Uri::from_parts(parts).unwrap();
let newreq = http::Request::get(&uri).body(()).unwrap();
let resolve_future = hyper_staticfile::resolve(&root, &newreq);
```

The new appproach would simply be: 
```
let mypath = Path::new(filename);
let resolve_future = hyper_staticfile::resolve(&root, mypath); 
```
We could take it a step further and make the resolve function into a `Resolver` struct which can be initialized with `&root`  so that we only have to run 
```
   let mypath = Path::new(filename);
   let resolve_future = my_resolver::resolve(mypath);
```
Thoughts?

**Comment by stephank** [comment]
Thanks for the feedback!

Maybe the immediate thing we can do (in a patch release, without breaking any code) is simply split off a `resolve_path` from the `resolve` function? Specifically, this part: https://github.com/stephank/hyper-staticfile/blob/92177f2a2246d68cd7000ed1e674a61269f54881/src/resolve.rs#L160-L171

Then the `resolve` API can stay the same, and internally delegate that part to `resolve_path`.

(The checks currently done in `resolve` have to be done somewhere, in regular file serving. Even if we remove them from `resolve`, that actually means moving them to `Static`.)

**Comment by stephank** [comment]
I've released hyper-staticfile 0.5 for hyper 0.13, and it also contains a `resolve_path`, so I'll close this issue now. :)
